pkg/blueprint/filesystem_customizations.go

```diff
@@ -24,17 +24,17 @@ func (fsc *FilesystemCustomization) UnmarshalTOML(data interface{}) error {
 		return fmt.Errorf("TOML unmarshal: mountpoint must be string, got %v of type %T", d["mountpoint"], d["mountpoint"])
 	}
 
-	switch d["size"].(type) {
+	switch d["minsize"].(type) {
```

Comment:
It seems part of the issue why this happend is that the `size`->`minsize` change happend in `FilesystemCustomization` but not in the unmarshal code. By adding a test that checks that marshal/unmarshal are symetric this can be prevented. I added a patch 
[0001-blueprint-add-marshal-unmarshal-symetry-test.patch.txt](https://github.com/user-attachments/files/16834969/0001-blueprint-add-marshal-unmarshal-symetry-test.patch.txt) with this idea.